src/components/ThemeProvider.tsx

```diff
@@ -14,6 +15,10 @@ function ThemeProvider({children, theme: staticThemePreference}: ThemeProviderPr
 
     const theme = useMemo(() => themes[themePreference], [themePreference]);
 
+    useEffect(() => {
+        DomUtils.addCSS(DomUtils.getAutofilledInputStyle(theme.text), 'autofill-input');
+    }, [theme.text]);
```

Comment:
This side-effect does not always work as expected as the `ThemeProvider` is used more than once (Main staticThemePreference=undefined and SignIn staticThemePreference=dark). This caused the app to have the incorrect autofill css as explained in the below rendering sequence:

- Main ThemeProvider (system)
- Main ThemeProvider (system) Adds Autofill CSS <-- (CSS A)
- SignIn ThemeProvider (dark)
- SignIn ThemeProvider (dark) Adds Autofill CSS <-- (CSS B)
- SignIn ThemeProvider (dark) is removed

The expected css is CSS A since the theme provider for CSS B is removed but that's not the cause. CSS B is still used.

Coming from https://github.com/Expensify/App/issues/51523